File: ios/keyChainWrapper.m

//
//  keyChainWrapper.m
//  Digital_Signature
//
//  Created by Wynne Tran on 2022-03-23.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(KeyChainWrapper, NSObject)
RCT_EXTERN_METHOD(savePassword:(NSString*)password callback: (RCTResponseSenderBlock)callback;)
@end

File: ios/keyChainWrapper.swift
//
//  keyChainWrapper.swift
//  Digital_Signature
//
//  Created by Wynne Tran on 2022-03-23.
//

import Foundation
import SwiftKeychainWrapper

@objc(KeyChainWrapper)
class KeyChainWrapper: NSObject {
  
  // delete warning error
  @objc static func requiresMainQueueSetup() -> Bool {
    return true;
  }
  
  @objc func savePassword(_ password:String, callback: RCTResponseSenderBlock){
    if(KeychainWrapper.standard.set(password, forKey: "password")){
      print( KeychainWrapper.standard.string(forKey: "password") as Any)
      callback([true])
      
    }
    else{
      callback([false])
      
    }
  }
}
